src/figuresClasses.ts

```diff
@@ -1,11 +1,87 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: string;
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    if (a <= 0 || b <= 0 || c < +0) {
+      throw new Error('Length of sides must be greater than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error(
+        'The longest side must be less or equal than sum of the other sides',
+      );
+    }
+  }
+
+  getArea(): number {
+    const semiPerimetr = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      semiPerimetr *
+        (semiPerimetr - this.a) *
+        (semiPerimetr - this.b) *
+        (semiPerimetr - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  public shape: string;
 
-export class Triangle implements Figure {}
+  constructor(
```

Comment:
[CODE KNOWLEDGE] The manual assignment of 'this.shape' in the constructor is unnecessary because TypeScript can automatically create and assign class properties when using public in the constructor parameters. You can remove this assignment.